### PR TITLE
Replace log.Fatal by return errors in deploy.go

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Return errors instead of doing `log.Fatal` inside functions

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: cmd/deploy.go [hunk 1]
```diff
@@ -159,20 +159,21 @@ func (dc *deployCmd) load(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("failed to get client: %s", err.Error())
 	}
 
-	// autofillApimodel calls log.Fatal() directly and does not return errors
```
Comment (CecileRobertMichon) [inline cmd/deploy.go]: not anymore 😈

File: cmd/deploy_test.go [hunk 2]
```diff
@@ -222,7 +222,11 @@ func TestAutoSufixWithDnsPrefixInApiModel(t *testing.T) {
 
 		client: &armhelpers.MockACSEngineClient{},
 	}
-	autofillApimodel(deployCmd)
+
+	err = autofillApimodel(deployCmd)
+	if err != nil {
+		t.Fatalf("unexpected error autofilling the example apimodel: %s", err)
```
Comment (jackfrancis) [inline cmd/deploy_test.go]: This is the last line of defense? (i.e., no return err?)

Comment (CecileRobertMichon) [reply cmd/deploy_test.go]: it's a unit test

Comment (jackfrancis) [reply cmd/deploy_test.go]: /me takes off beer goggles

File: cmd/deploy_test.go [hunk 3]
```diff
@@ -423,25 +439,28 @@ func testAutodeployCredentialHandling(t *testing.T, useManagedIdentity bool, cli
...
+	}
 
 	// cleanup, since auto-populations creates dirs and saves the SSH private key that it might create
 	defer os.RemoveAll(deployCmd.outputDirectory)
 
 	cs, _, err = validateApimodel(apiloader, cs, ver)
 	if err != nil {
-		log.Fatalf("unexpected error validating apimodel after populating defaults: %s", err)
```
Comment (CecileRobertMichon) [inline cmd/deploy_test.go]: @jackfrancis you actually just made me realize this should be `t` not `log`

Comment (CecileRobertMichon) [reply cmd/deploy_test.go]: so I changed it

Comment (jackfrancis) [reply cmd/deploy_test.go]: O.K. beer googles worked!

Comment (CecileRobertMichon) [reply cmd/deploy_test.go]: 🍻

**Pull request merged**